R/gamlss-tidiers.R

```diff
@@ -43,8 +57,25 @@ tidy.gamlss <- function(x, quick = FALSE, ...) {
   nn <- c("estimate", "std.error", "statistic", "p.value")
   ret <- fix_data_frame(s, nn)
 
-  # add parameter types. This assumes each coefficient table starts
-  # with "(Intercept)": unclear if this is guaranteed
-  parameters <- x$parameters[cumsum(ret$term == "(Intercept)")]
+  if (conf.int) {
+    cilist <- lapply(
+      x$parameters,
+      function(w) confint(x, what = w, level = conf.level, ...)
+    )
+    cimat <- do.call(rbind, cilist)
+    ret <- bind_cols(ret, conf.low = cimat[, 1], conf.high = cimat[, 2])
+  }
+  ## add parameter types
+  coefs <- x[paste0(x$parameters, ".coefficients")]
+  parameters <- rep(
+    x$parameters,
+    vapply(coefs, length, 1L)
+  )
   bind_cols(parameter = parameters, ret)
 }
+
+#' @export
+glance.gamlss <- function(x, ...) {
+  ret <- tibble(df = x$df.fit)
+  finish_glance(ret, x)
```

Comment:
`finish_glance()` has also been removed from `broom` so you'll need to recreate whatever specific functionality from that that's getting used here.